buildSrc/src/main/java/org/elasticsearch/gradle/Version.java

```diff
@@ -16,35 +16,43 @@
     /**
      * Suffix on the version name.
      */
-    private final String suffix;
+    private final String qualifier;
 
     private static final Pattern pattern =
             Pattern.compile("(\\d)+\\.(\\d+)\\.(\\d+)(-alpha\\d+|-beta\\d+|-rc\\d+)?(-SNAPSHOT)?");
 
-    public Version(int major, int minor, int revision, String suffix, boolean snapshot) {
+    Version(int major, int minor, int revision, String qualifier, boolean snapshot) {
         Objects.requireNonNull(major, "major version can't be null");
         Objects.requireNonNull(minor, "minor version can't be null");
         Objects.requireNonNull(revision, "revision version can't be null");
         this.major = major;
         this.minor = minor;
         this.revision = revision;
         this.snapshot = snapshot;
-        this.suffix = suffix == null ? "" : suffix;
+        if (qualifier == null || qualifier.isEmpty()) {
```

Comment:
Why do we need this leniency? The semantics should be either no qualifier is null, or empty string, but not either.

Reply:
I think it's because the instance was constructed with null most of the time but, we mostly get the qualifier to compare against it, so having `""` instead  is more convenient and readable. 
We could disallow nulls in the constructor, I would clean this up as part of  #32872